tests/admin_views/tests.py

```diff
@@ -6181,3 +6181,36 @@ def test_implicitly_generated_pk(self):
         post_data = {'name': '2'}
         response = self.client.post(reverse('admin:admin_views_implicitlygeneratedpk_change', args=(1,)), post_data)
         self.assertEqual(response.status_code, 302)
+
+
+@override_settings(ROOT_URLCONF='admin_views.urls')
+class AddM2MFieldsToObjChangeHistoryTest(TestCase):
+    """
+    #27998 - If m2m fields are changed on an object those fields should be
+    listed in the change message logged in the object's history.
+    """
+
+    @classmethod
+    def setUpTestData(cls):
+        cls.superuser = User.objects.create_superuser(username='super', password='secret', email='super@example.com')
+
+    def setUp(self):
+        self.client.force_login(self.superuser)
+
+    def test_change_m2m_field_logentry(self):
+        user = User.objects.create_user(username='fredsmith', password='supersekret')
+        group = Group.objects.create(name='newgroup')
+        post_data = {
+                'username': user.username,
+                'groups': ['1'],
+                'is_active': 'on',
+                'date_joined_0': str(user.date_joined.date()),
+                'date_joined_1': str(user.date_joined.time().strftime('%X')),
+                'initial-date_joined_0': str(user.date_joined.date()),
+                'initial-date_joined_1': str(user.date_joined.time().strftime('%X')),
+                '_save': '1'
+                }
+        self.client.post(reverse('admin:auth_user_change', args=(user.pk,)), post_data)
+        user_ctype = ContentType.objects.get_for_model(User)
+        log = LogEntry.objects.filter(content_type=user_ctype, object_id=user.pk).last()
+        self.assertEqual(log.get_change_message(), 'Changed groups.')
```

Comment:
@timgraham It might be more appropriate in another commit then. I believe I wanted to make sure nothing was logged if a m2m backed inline was submitted without changes.